src/composables/trade/useTokenApproval.ts

```diff
@@ -90,14 +93,22 @@ export default function useTokenApproval(tokenInAddress, amount, tokens) {
       const [tx] = await approveTokens(auth.web3, config.addresses.vault, [
         tokenInAddress.value
       ]);
-      approvalTxListener(tx.hash);
+      txHandler(tx);
     } catch (e) {
       console.log(e);
       approving.value = false;
     }
   }
 
-  function approvalTxListener(hash: string) {
+  function txHandler(tx: TransactionResponse): void {
+    if (supportsBlocknative.value) {
+      blocknativeTxHandler(tx.hash);
+    } else {
+      ethersTxHandler(tx);
+    }
+  }
+
+  function blocknativeTxHandler(hash: string): void {
```

Comment:
This is inconsistent with what we have in the `InvestForm` where both handlers receive a `TransactionResponse` and BN pulls out the hash internally.